File: lista_de_exercicios/questao4/questao4.go
package main

import "fmt"

type Animal interface {
    Som() string
}

type Cachorro struct{}

func (c Cachorro) Som() string {
    return "Latido"
}

type Gato struct{}

func (g Gato) Som() string {
    return "Miau"
}

func main() {
    var cachorro Animal = Cachorro{}
    var gato Animal = Gato{}
    fmt.Println(cachorro.Som())  // Latido
    fmt.Println(gato.Som())      // Miau
}

File: lista_de_exercicios/questao4/questao4.py
class Animal:
    def som(self):
        pass

class Cachorro(Animal):
    def som(self):
        return "Latido"

class Gato(Animal):
    def som(self):
        return "Miau"

# Exemplo de uso
cachorro = Cachorro()
gato = Gato()
print(cachorro.som())  # Latido
print(gato.som())      # Miau
